onnxruntime/contrib_ops/webgpu/quantization/gather_block_quantized.cc

```diff
@@ -38,17 +38,23 @@ Status GatherBlockQuantizedProgram::GenerateShaderCode(ShaderHelper& shader) con
     shader.MainFunctionBody()
         << "let indices_indices = " << output.IndicesGet("output_indices", "uniforms.gather_axis") << ";\n";
   }
+
+  shader.MainFunctionBody()
+      << "var index = " << indices.GetByIndices("indices_indices") << ";\n"
+      << "if (index < 0) { index += " << x_shape_[gather_axis_] << ";}\n"
```

Comment:
Not related to this PR (because this part is not changed):

The current implementation of `GatherBlockQuantizedProgram::GenerateShaderCode` seems to have dependency on `X`'s shape (value of `x_shape_[gather_axis_]` is directly written into the shader) but the value of `x_shape_[gather_axis_]` is not a part of cache hint. It looks to me like this may potentially cause an inconsistency when a new input `X` with different shape reuses the same program.